### PR TITLE
#M19. Simplify and speed-up rendering scores

### PR DESCRIPTION
**Pull request by stefdworschak**
## Description
- Use custom SQL query to pre-aggregate scores instead of looping through all scores
- Use pandas to speed up calculations

Accidentially closed previous PR: https://github.com/Code-Institute-Community/ci-hackathon-app/pull/177

## Pull request type

- [x] #M01 (Miscellaneous User Story #01)
- [ ] #P02 (Participant User Story #02)
- [ ] #S03 (Staff User Story #03)
- [ ] #A04 (Admin User Story #04)
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue
#176

## Configuration instructions
- Update requirements using `pip install -r requirements.txt`

## Testing
- All tests passing (apart from broken tests for the custom Slack provider which have not been working from the start

### CODE REVIEW

**Comment by stefdworschak** [comment]
Addressed all changes made in PR: #177 
Merging this PR.

File: hackathon/templates/hackathon/final_score.html [hunk 1]
```diff
@@ -11,77 +11,34 @@
...
-<div class="card mb-3">
+{% if scores_table %}
+<div class="card mb-3 judge-scores-card">
     <table class="table table-bordered final_score_table w-100 m-0 p-0">
         <thead>
             <tr class="table_titles">
-                <th scope="col" class="text-left"> Team / Project
-                    <i class="fas fa-info-circle ml-2" title="Teams with no projects yet:{{teams_without_projects}}"></i>
+                {% for header in scores_table.headers %}
+                {% if forloop.counter == 1%}
```
Comment (TravelTimN) [inline hackathon/templates/hackathon/final_score.html:20]: Why not simplify this to:
`{% if forloop.first %}`

File: hackathon/templates/hackathon/final_score.html [hunk 2]
```diff
@@ -11,77 +11,34 @@
...
-                {% with categories=category_scores.values|to_list|first %}
-                {% for category in categories.keys %}
-                {% if category != 'project_name' %}
-                <th scope="col" >
-                    {{category}}
-                </th>
-                {% endif %}
+                {% for col in row %}
+                <!-- class="not-counted-scores" -->
+                <td {% if forloop.counter == 1%}class="text-left"{% elif forloop.last %} scope="row"{% endif %}>{{col}}</td>
```
Comment (TravelTimN) [inline hackathon/templates/hackathon/final_score.html:35]: Same here...?
`{% if forloop.first %}`
since you're already using `{% if forloop.last %}`

**Review by TravelTimN — COMMENTED** [review]
Interesting concept to manage the judging scores (and long load times to calculate).
I recently heard of Pandas, and got me curious about it.
Looks great! 👍🏻 
(only suggestions below is just recommendation, not requirement.... simplifying the conditional Jinja checks slightly)

**Pull request merged**